app/models/grda_warehouse/warehouse_reports/youth/homeless_youth_report.rb

```diff
@@ -0,0 +1,466 @@
+module GrdaWarehouse::WarehouseReports::Youth
+  class HomelessYouthReport
+
+    def initialize(filter)
+      @start_date = filter.start
+      @end_date = filter.end
+    end
+
+    # A. Core Services
+
+    def new_enrollments_scope
+      GrdaWarehouse::YouthIntake::Base.
+        served.
+        open_between(start_date: @start_date, end_date: @end_date).
+        open_after(@start_date)
+    end
+
+    def section_1_scope
+      GrdaWarehouse::YouthIntake::Base.
+        served.
+        open_between(start_date: @start_date, end_date: @end_date).
+        where(street_outreach_contact: "Yes")
+    end
+
+    def one_a
+      @one_a ||= get_client_ids(section_1_scope.
+        where(housing_status: homeless_statuses))
+    end
+
+    def one_b
+      @one_b ||= get_client_ids(section_1_scope.
+        where(housing_status: 'At risk of homelessness'))
+    end
+
+    def section_2_scope
+      GrdaWarehouse::YouthIntake::Base.
+        served.
+        open_between(start_date: @start_date, end_date: @end_date).
+        where(street_outreach_contact: "No")
+    end
+
+    def two_a
+      @two_a ||= get_client_ids(section_2_scope.
+        where(housing_status: homeless_statuses))
+    end
+
+    def two_b
+      @two_b ||= get_client_ids(section_2_scope.
+        where(housing_status: 'At risk of homelessness'))
+    end
+
+    def two_c
+      @two_c ||= {}
```

Comment:
This is still going to execute lines 54-57 every time `two_c` is called.  Is that the intention?  I'm thinking maybe there should be a begin/end here so that the calculation is cached.